lib/Backend/arm64/ARM64Encoder.h

```diff
@@ -1921,12 +1921,12 @@ EmitBitfieldCommon(
     Arm64CodeEmitter &Emitter,
     Arm64SimpleRegisterParam Dest,
     Arm64SimpleRegisterParam Src,
-    ULONG Immr,
-    ULONG Imms,
+    ULONG64 Immr,
+    ULONG64 Imms,
     ULONG Opcode
     )
 {
-    return Emitter.EmitFourBytes(Opcode | (Immr << 16) | (Imms << 10) | (Src.RawRegister() << 5) | Dest.RawRegister());
+    return Emitter.EmitFourBytes(Opcode | (ULONG(Immr) << 16) | (ULONG(Imms) << 10) | (Src.RawRegister() << 5) | Dest.RawRegister());
```

Comment:
Will truncating cause data loss?

Reply:
Not for this case, no. Upstream functions ensure it won't happen. I just switched to 64-bit parameters here to prevent excessive warnings, and there's no perf difference.